app/kuma-dp/pkg/dataplane/readiness/component.go

```diff
@@ -24,34 +23,28 @@ const (
 
 // Reporter reports the health status of this Kuma Dataplane Proxy
 type Reporter struct {
-	socketPath    string
-	isTerminating atomic.Bool
+	localListenAddr string
+	localListenPort uint32
+	isTerminating   atomic.Bool
 }
 
 var logger = core.Log.WithName("readiness")
 
-func NewReporter(socketPath string) *Reporter {
+func NewReporter(localIPAddr string, localListenPort uint32) *Reporter {
 	return &Reporter{
-		socketPath: socketPath,
+		localListenPort: localListenPort,
+		localListenAddr: localIPAddr,
 	}
 }
 
 func (r *Reporter) Start(stop <-chan struct{}) error {
-	_, err := os.Stat(r.socketPath)
-	if err == nil {
-		// File is accessible try to rename it to verify it is not open
-		newName := r.socketPath + ".bak"
-		err = os.Rename(r.socketPath, newName)
-		if err != nil {
-			return errors.Errorf("file %s exists and probably opened by another kuma-dp instance", r.socketPath)
-		}
-		err = os.Remove(newName)
-		if err != nil {
-			return errors.Errorf("not able the delete the backup file %s", newName)
-		}
+	protocol := "tcp"
+	addr := r.localListenAddr
+	if govalidator.IsIPv6(addr) {
+		protocol = "tcp6"
+		addr = fmt.Sprintf("[%s]", addr)
```

Comment:
nitpick: just FYI you could use `netip.AddrPort` and have it handle formatting, but it probably won't blow up anywhere so we can leave it like that